src/math_tools/spline_2d_partial_derivative.hpp

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "ddc_alias_inline_functions.hpp"
+#include "ddc_aliases.hpp"
+#include "ipartial_derivative.hpp"
+#include "spline_builder_2d_cache.hpp"
+
+
+/**
+ * @brief A class which implements a partial derivative operator
+ * using a 2d spline interpolation.
+ * @tparam SplineBuilder2D A 2D spline builder.
+ * @tparam SplineEvaluator2D A 2D spline evaluator.
+ */
+template <class SplineBuilder2DCache, class SplineEvaluator2D, class DerivativeDimension>
+class Spline2DPartialDerivative
+    : public IPartialDerivative<
+              typename SplineEvaluator2D::batched_evaluation_domain_type,
+              DerivativeDimension>
+{
+    static_assert(
+            std::is_same_v<
+                    typename SplineEvaluator2D::continuous_dimension_type1,
+                    DerivativeDimension> || std::is_same_v<typename SplineEvaluator2D::continuous_dimension_type2, DerivativeDimension>);
```

Comment:
The clang indenting tends to display better with extra brackets when you have `||` and `<>` in an expression
```suggestion
    static_assert(
            (std::is_same_v<
                    typename SplineEvaluator2D::continuous_dimension_type1,
                    DerivativeDimension>) || (std::is_same_v<typename SplineEvaluator2D::continuous_dimension_type2, DerivativeDimension>));
```